codegen/facelift/templates/IPCProxyAdapter.template.cpp

```diff
@@ -55,76 +51,132 @@
     {% endif %}
 }
 
-void {{className}}::deserializePropertyValues(InputIPCMessage &msg, bool isCompleteSnapshot)
+void {{className}}::unmarshalProperties(const QVariantMap& values)
 {
-    {% for property in interface.properties %}
-    {% if property.type.is_interface %}
-    bool emit_{{property.name}}ChangeSignal = false;
-    QString {{property.name}}_objectPath;
-    if (deserializeOptionalValue(msg, {{property.name}}_objectPath, isCompleteSnapshot))
-    {
-        m_{{property.name}}Proxy.update({{property.name}}_objectPath);
-        m_{{property.name}} = m_{{property.name}}Proxy.getValue();
-        emit_{{property.name}}ChangeSignal = true;
+    QMap<QString, bool> emitChangeSignal;
```

Comment:
There is no need to use a QMap here. keep the existing approach with one local bool variable per property.

Reply:
When a proxy successfully connects to its server counterpart, some application code may legitimately assume that the "readyChanged" signal is triggered after all other property change signals, since "readyChanged with ready==true" means that the proxy is fully initialised and ready to be used.

Reply:
good point.